internal/dag/cache.go

```diff
@@ -64,7 +66,8 @@ type KubernetesCache struct {
 
 // init creates the internal cache storage. It is called implicitly from the public API.
 func (kc *KubernetesCache) init() {
-	kc.ingresses = make(map[types.NamespacedName]*v1beta1.Ingress)
+	kc.ingresses = make(map[types.NamespacedName]*networking_v1.Ingress)
+	kc.ingressclasses = make(map[string]*networking_v1.IngressClass)
```

Comment:
it's still not clear to me that we need to cache `IngressClass`, but we can always remove it later if we don't need it.